### PR TITLE
Add doc for PA:CATALOG_OBJECT variable params: kind and contentType filters

### PR DESCRIPTION
**Pull request by tao-xinxiu**
These parameters (kind and contentType filters) are optional

### CODE REVIEW

**Review by AlexFalessi — APPROVED** [review]

**Review by marounkoussaifi — APPROVED** [review]

File: src/docs/user/references/Variables.adoc [hunk 1]
```diff
@@ -145,9 +145,12 @@ Examples: PA:MODEL_FROM_URL(file:///srv/machines_list_model.txt), if the file ma
...
-* *PA:CATALOG_OBJECT* : variable syntax must be a valid expression that matches the following pattern: `bucketName/objectName[/revision]`.
+* *PA:CATALOG_OBJECT*, *PA:CATALOG_OBJECT(kind)*, *PA:CATALOG_OBJECT(,contentType)*, *PA:CATALOG_OBJECT(kind,contentType)* : variable syntax must be a valid expression that matches the following pattern: `bucketName/objectName[/revision]`.
 Note that the revision sub-pattern is a hash code number represented by 13 digit. +
+A filter can be specified in the PA:CATALOG_OBJECT model. In that case, the variable value must be a catalog object which matches the kind and content type requirements.
+The scheduler server verifies that the object exists in the catalog and fulfills the requirements during workflow submission. +
 Examples: PA:CATALOG_OBJECT will accept "bucket-example/object-example/1539310165443" and "bucket-example/object-example" but not "bucket-example/object-example/153931016" (invalid revision number) and "bucket-example/" (missing object name).
+PA:CATALOG_OBJECT(Script,text/x-python) will only accept catalog objects which is a Python script but not a workflow object or a Groovy script.
```
Comment (mboussaa) [inline src/docs/user/references/Variables.adoc:153]: will only accept a catalog object which is 
or 
will only accept catalog objects which are Python scripts

File: src/docs/user/references/Variables.adoc [hunk 2]
```diff
@@ -145,9 +145,12 @@ Examples: PA:MODEL_FROM_URL(file:///srv/machines_list_model.txt), if the file ma
...
 Examples: PA:CRON will accept "5 * * * \*" but not "* * * *" (missing minutes sub-pattern).
 * *PA:SPEL(SpEL expression)* : variable syntax will be evaluated by a https://docs.spring.io/spring/docs/current/spring-framework-reference/html/expressions.html[SpEL] expression. Refer to next paragraph.
-* *PA:CATALOG_OBJECT* : variable syntax must be a valid expression that matches the following pattern: `bucketName/objectName[/revision]`.
+* *PA:CATALOG_OBJECT*, *PA:CATALOG_OBJECT(kind)*, *PA:CATALOG_OBJECT(,contentType)*, *PA:CATALOG_OBJECT(kind,contentType)* : variable syntax must be a valid expression that matches the following pattern: `bucketName/objectName[/revision]`.
 Note that the revision sub-pattern is a hash code number represented by 13 digit. +
+A filter can be specified in the PA:CATALOG_OBJECT model. In that case, the variable value must be a catalog object which matches the kind and content type requirements.
+The scheduler server verifies that the object exists in the catalog and fulfills the requirements during workflow submission. +
```
Comment (mboussaa) [inline src/docs/user/references/Variables.adoc:151]: "during workflow submission..."
is the check done even continuously before clicking on the check button?

**Review by mboussaa — APPROVED** [review]
lgtm: small remarks

File: src/docs/user/references/Variables.adoc (hunk 2 above)
Comment (tao-xinxiu) [reply src/docs/user/references/Variables.adoc]: No, actually I meant that before we click the check or submit button, it's not checked (e.g., during workflow design). I've changed it to "when the workflow is submitted". Thank you.

File: src/docs/user/references/Variables.adoc [hunk 3]
```diff
@@ -145,9 +145,12 @@ Examples: PA:MODEL_FROM_URL(file:///srv/machines_list_model.txt), if the file ma
...
-* *PA:CATALOG_OBJECT* : variable syntax must be a valid expression that matches the following pattern: `bucketName/objectName[/revision]`.
+* *PA:CATALOG_OBJECT*, *PA:CATALOG_OBJECT(kind)*, *PA:CATALOG_OBJECT(,contentType)*, *PA:CATALOG_OBJECT(kind,contentType)* : variable syntax must be a valid expression that matches the following pattern: `bucketName/objectName[/revision]`.
 Note that the revision sub-pattern is a hash code number represented by 13 digit. +
+A filter can be specified in the PA:CATALOG_OBJECT model. In that case, the variable value must be a catalog object which matches the kind and content type requirements.
+The scheduler server verifies that the object exists in the catalog and fulfills the requirements when the workflow is submitted. +
 Examples: PA:CATALOG_OBJECT will accept "bucket-example/object-example/1539310165443" and "bucket-example/object-example" but not "bucket-example/object-example/153931016" (invalid revision number) and "bucket-example/" (missing object name).
+PA:CATALOG_OBJECT(Script,text/x-python) will only accept a catalog object which is a Python script but not a workflow object or a Groovy script.
```
Comment (fviale) [inline src/docs/user/references/Variables.adoc:153]: I can't remember, is this case sensitive or case insensitive ?

Comment (tao-xinxiu) [reply src/docs/user/references/Variables.adoc:153]: Case insensitive.

Comment (fviale) [reply src/docs/user/references/Variables.adoc:153]: Maybe it's worth mentioning it ?

**Review by fviale — APPROVED** [review]

**Pull request merged**